### PR TITLE
update roledefinition filters

### PR DESCRIPTION
**Pull request by darshanhs90**
Linked to https://github.com/Azure/azure-rest-api-specs/pull/2586
waiting for https://github.com/Azure/azure-sdk-for-net/pull/4079 to be merged

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/Authorization/Authorization.Tests/Tests/BasicTests.cs [hunk 1]
```diff
@@ -687,8 +687,56 @@ public void RoleDefinitionsByIdTests()
             }
         }
 
-        //[Fact(Skip = "After upgrade to vs2017, starts failing. Needs investigation")]
-        [Fact]
+		[Fact]
+		public void RoleDefinitionsFilterTests()
```
Comment (dsgouda) [inline src/SDKs/Authorization/Authorization.Tests/Tests/BasicTests.cs]: nit: FIx indentation

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please pull down the latest changes and run:
-  `msbuild build.proj /t:build /p:Scope=SDKs\Monitor
-  generate.cmd` to generate the code
These steps should generate the `.props` and `.txt` files, you need to commit these files

**Comment by dsgouda** [comment]
There are some build failures, please address them

**Comment by darshanhs90** [comment]
.props file isnt changed,since this isnt introducing a new api version

**Comment by darshanhs90** [comment]
@dsgouda Fixed the build issues

**Review by dsgouda — APPROVED** [review]
LGTM subject to successful CI builds, @shahabhijeet PTAL and merge

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**